docs/guides/add-onboarding-flow.mdx

```diff
@@ -0,0 +1,232 @@
+---
+title: How to Add an Onboarding Flow for your Application
+description: Leverage Clerk’s customizable session tokens, publicMetadata and Next’s Middleware to create a robust onboarding experience within a few lines of code.
+---
+
+# How to Add an Onboarding Flow for your Application
+
+As part of your onboarding flow, you may want to collect extra information from your user and use it to drive your application state. Let’s walk through a quick example using Next.js and TypeScript to show you how simple implementing an onboarding flow can be.
+
+In this guide, you will learn how to:
+
+1. Add custom claims to your session token
+2. Configure your middleware to read session data
+3. Update with the user’s onboarding state
+
+To see a working example, check out our sample demonstration app [here](https://sample-onboarding-app.clerkpreview.com/).
+
+<Callout type="info" emoji="💡">
+  The examples below have been pared down to the bare minimum to enable you to
+  easily customize them to your needs, you can build them with the [Clerk + Next
+  Quickstart](https://clerk.com/docs/quickstarts/nextjs) using @clerk/nextjs
+  4.29.5 and Next 14.0.4.
+</Callout>
+
+Let’s get started!
+
+<img />
+
+## Add custom claim to your session token
+
+Session tokens are JWTs that are generated by Clerk on behalf of your instance, and contain claims that allow you to store data about a user’s session. With Clerk, when a session token exists for a user, it indicates that the user is authenticated, and the associated claims can be retrieved at any time. [Learn More](https://clerk.com/docs/backend-requests/making/custom-session-token)
+
+First, navigate to [Sessions](https://dashboard.clerk.com/last-active/sessions?_gl=1*1bx2kkm*_gcl_au*MTcwMzAwNTE2LjE3MDQ5MTA2MjAuMTQ0NTAzMDk3Ny4xNzA1MDM3Njk5LjE3MDUwMzc3MDU.*_ga*OTkyMTU1NTM2LjE2ODk0NzAzMDU.*_ga_1WMF5X234K*MTcxMjE5NDgyMC40OS4xLjE3MTIxOTU0ODQuMC4wLjA.) in your Clerk Dashboard and click the ‘Edit’ button. In the modal that opens, there will be a window where you can augment your session token with custom claims.
+
+In there, add the following and hit save:
+
+```json
+{
+  "metadata": "{{user.public_metadata}}"
+}
+```
+
+<img />
+
+If you haven’t already, we can make the public metadata type information accessible to our application by adding the following to src/types/globals.d.ts:
+
+```ts
+export {};
+
+declare global {
+  interface CustomJwtSessionClaims {
+    metadata: {
+      onboardingComplete?: boolean;
+    };
+  }
+}
+```
+
+We have just added custom data to our session token in the Clerk Dashboard and made those claims accessible to our app. Next, we’ll use clerkMiddleware to redirect the user based on `onboardingComplete` status.
```

Comment:
I still don't think this should be here. In a blog maybe, but in docs, this is redundant - the user can simply go to the next step to understand what they're going to do next.

Reply:
I remove this text that's at the end of every step because it's redundant and informal. The user can simply move on to the next step to see what they need to do next. Maybe in blogs this is okay, because blogs are more informal and the writer speaks directly to the user more, but in technical writing, this is not proper prose.